File: publlic_html/python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  silnia.py

def silnia_it(n):
    # 0! = 1
    # n! = 1 * 2 * ... * n
    wynik = 1
    for i in range(1, n + 1):
        print(wynik)
        wynik = wynik * i
    return wynik
    
def main(args):
    # zmienne lokalne
    n = int(input("Podaj liczbę: "))
    print("{}! = {}". format(n, silnia_it(n)))
   
    return 0

if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))

File: klasa 2/cpp/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

float silnia_it(int n)
{
    int silnia = 1;
    int i;
    for (i = 1; i <= n; i++)
    {
        silnia = silnia * i;
    }
        return silnia;
}

float silnia_re(int n)
{
    if (n == 0) return 1;
    return silnia_re(n-1) * n;
}

int main(int argc, char **argv)
{
    int n;
    cout << "Podaj liczbę: ";
    cin >> n;
    
    cout << "Wynik: " << silnia_re(n) << endl;
    cout << "Wynik: " << silnia_it(n) << endl;
    
    
    
	return 0;
}
